src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb

```diff
@@ -593,25 +593,28 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             Dim node = root.FindToken(position).Parent
             While node IsNot Nothing
-                If TypeOf node Is MethodBlockBaseSyntax AndAlso Not TypeOf node.Parent Is PropertyBlockSyntax Then
-                    Return node
-                End If
+                If useFullSpan OrElse node.Span.Contains(position) Then
 
-                If TypeOf node Is PropertyStatementSyntax AndAlso Not TypeOf node.Parent Is PropertyBlockSyntax Then
-                    Return node
-                End If
+                    If TypeOf node Is MethodBlockBaseSyntax AndAlso Not TypeOf node.Parent Is PropertyBlockSyntax Then
+                        Return node
+                    End If
 
-                If TypeOf node Is EventStatementSyntax AndAlso Not TypeOf node.Parent Is EventBlockSyntax Then
-                    Return node
-                End If
+                    If TypeOf node Is PropertyStatementSyntax AndAlso Not TypeOf node.Parent Is PropertyBlockSyntax Then
+                        Return node
+                    End If
 
-                If TypeOf node Is PropertyBlockSyntax OrElse
-                   TypeOf node Is TypeBlockSyntax OrElse
-                   TypeOf node Is EnumBlockSyntax OrElse
-                   TypeOf node Is NamespaceBlockSyntax OrElse
-                   TypeOf node Is EventBlockSyntax OrElse
-                   TypeOf node Is FieldDeclarationSyntax Then
-                    Return node
+                    If TypeOf node Is EventStatementSyntax AndAlso Not TypeOf node.Parent Is EventBlockSyntax Then
+                        Return node
+                    End If
+
+                    If TypeOf node Is PropertyBlockSyntax OrElse
+                       TypeOf node Is TypeBlockSyntax OrElse
+                       TypeOf node Is EnumBlockSyntax OrElse
+                       TypeOf node Is NamespaceBlockSyntax OrElse
+                       TypeOf node Is EventBlockSyntax OrElse
+                       TypeOf node Is FieldDeclarationSyntax Then
```

Comment:
Note that I didn't write this function, just indented it try &w=0 in the diff :smile:.

However, since they don't have bodies, I'm totally find with saying they aren't a "Block".  AFAIK, this interface is only used to populate the current block scope in the Find in Files dialog, so I'm not *terribly* about things like this.